Utils.cpp

```diff
@@ -5,226 +5,221 @@
+            indexes.emplace_back(i);
+            prices.push_back((INT_TYPE) trade_prices[i]);
+        }
+
+    std::queue<INT_TYPE> s{};
+
+    for (int j = 0; j < prices.size(); ++j) {
+        INT_TYPE count = 0;
+        INT_TYPE p = prices[j];
+        INT_TYPE t = indexes[j];
+
+        int_vector lstp(trade_prices.begin(), std::next(trade_prices.begin(), t));
+        INT_TYPE sz = lstp.size();
+
+        s.push(0);
+
+        while (sz > count++) {
+            INT_TYPE lp = lstp[sz - count];
+            if (p > lp) {
+                s.back() = 1;
+                break;
+            }
+            if (p < lp) {
+                s.back() = -1;
+                break;
+            }
+        }
+    }
+
+    for (int i = 0; i < res.step.size(); ++i)
+        if (res.step[i] == 0) {
+            res.initiator[i] = s.front();
+            s.pop();
+        }
 }
 
 fp_vector
 get_midpoint(
-	const fp_vector &ask_time,
-	const fp_vector &bid_time,
-	const fp_vector &ask_price,
-	const fp_vector &bid_price,
-	const fp_vector &trades_time)
-{
-	fp_vector ask = get_lastquote(ask_time, ask_price, trades_time);
-	fp_vector bid = get_lastquote(bid_time, bid_price, trades_time);
-	fp_vector midpoint(ask.size());
-
-	for (int i = 0; i < midpoint.size(); ++i)
-		if (ask[i] < bid[i])
-			midpoint[i] = fp_nan();
-		else
-			midpoint[i] = (ask[i] + bid[i]) / 2.0;
-
-	return midpoint;
+        const fp_vector &ask_time,
+        const fp_vector &bid_time,
+        const fp_vector &ask_price,
+        const fp_vector &bid_price,
+        const fp_vector &trades_time) {
+    fp_vector ask = get_lastquote(ask_time, ask_price, trades_time);
+    fp_vector bid = get_lastquote(bid_time, bid_price, trades_time);
+    fp_vector midpoint(ask.size());
+
+    for (int i = 0; i < midpoint.size(); ++i)
+        if (ask[i] < bid[i])
+            midpoint[i] = fp_nan();
+        else
+            midpoint[i] = (ask[i] + bid[i]) / 2.0;
+
+    return midpoint;
 }
 
 fp_vector
 get_lastquote(
-	const fp_vector &quotes_time,
-	const fp_vector &quotes_price,
-	const fp_vector &trades_time)
-{
-	fp_vector last_quote(trades_time.size(), 0.0);
-	int_vector ind = vec_searchsorted(quotes_time, trades_time, Side::LEFT);
-
-	for (int i = 0; i < ind.size(); ++i)
-	{
-		--ind[i];
-		if (ind[i] < 0)
-			last_quote[i] = fp_nan();
-		else
-			last_quote[i] = quotes_price[ind[i]];
-	}
-
-	return last_quote;
+        const fp_vector &quotes_time,
+        const fp_vector &quotes_price,
+        const fp_vector &trades_time) {
+    fp_vector last_quote(trades_time.size(), 0.0);
+    int_vector ind = vec_searchsorted(quotes_time, trades_time, Side::LEFT);
+
+    for (int i = 0; i < ind.size(); ++i) {
+        --ind[i];
+        if (ind[i] < 0)
+            last_quote[i] = fp_nan();
+        else
+            last_quote[i] = quotes_price[ind[i]];
+    }
+
+    return last_quote;
 }
+
+fp_vector outerJoin(const fp_vector &lastask, const fp_vector &lastbid) {
```

Comment:
La o adica, doar ca sa fie consistent cu restul functiilor, am putea redenumi asta in `outer_join`. E cam singura functie acum cu `camelCase`.